src/base_object.h

```diff
@@ -28,6 +28,8 @@
 
 namespace node {
 
+int GenDebugSymbols();
```

Comment:
Correct me if I'm wrong but I don't think you have to forward-declare it, just the `friend` stanza inside the class is enough.